src/syntax/ast.rs

```diff
@@ -28,7 +31,6 @@ pub struct Pattern {
 
 #[derive(Debug, PartialEq)]
 pub enum Expression {
-    Pattern(Pattern),
     String(String),
     Number(Number),
```

Comment:
I was simply going by the ASDL and noticed that it was missing.  If you think it's not needed and that `StringExpression { value }` nodes can be added to JSON during the serialization, that's totally fine too.